controllers/helpers.go

```diff
@@ -299,45 +302,136 @@ func GetSubnetUUID(ctx context.Context, client *prismclientv3.Client, peUUID str
 }
 
 // GetImageUUID returns the UUID of the image with the given name
-func GetImageUUID(ctx context.Context, client *prismclientv3.Client, imageName, imageUUID *string) (string, error) {
-	var foundImageUUID string
-
-	if imageUUID == nil && imageName == nil {
+func GetImageUUID(
+	ctx context.Context,
+	client *prismclientv3.Client,
+	imageName,
+	imageUUID,
+	imageLookupFormat,
+	imageLookupBaseOS,
+	k8sVersion *string,
+) (string, error) {
+	if imageUUID == nil && imageName == nil && imageLookupBaseOS == nil {
 		return "", fmt.Errorf("image name or image uuid must be passed in order to retrieve the image")
 	}
+	// name serves as the highest precedence
+	if imageName != nil {
+		resolvedUUID, err := getImageByName(ctx, client, *imageName)
+		if err != nil {
+			return "", err
+		}
+		return resolvedUUID, nil
+	}
 	if imageUUID != nil {
-		imageIntentResponse, err := client.V3.GetImage(ctx, *imageUUID)
+		resolvedUUID, err := getImageByUUID(ctx, client, *imageUUID)
 		if err != nil {
-			if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
-				return "", fmt.Errorf("failed to find image with UUID %s: %v", *imageUUID, err)
-			}
+			return "", err
 		}
-		foundImageUUID = *imageIntentResponse.Metadata.UUID
-	} else { // else search by name
-		responseImages, err := client.V3.ListAllImage(ctx, "")
+		return resolvedUUID, nil
```

Comment:
Unfortunately, this function is replaced in https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/489. That is about to merge.

Reply:
rebased thanks for the heads up!